wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/CommandScheduler.java

```diff
@@ -140,15 +140,16 @@ public void clearButtons() {
    * @param requirements  The command requirements
    */
   private void initCommand(Command command, boolean interruptible, Set<Subsystem> requirements) {
-    command.initialize();
     CommandState scheduledCommand = new CommandState(interruptible);
     m_scheduledCommands.put(command, scheduledCommand);
-    for (Consumer<Command> action : m_initActions) {
-      action.accept(command);
-    }
+    command.initialize();
     for (Subsystem requirement : requirements) {
       m_requirements.put(requirement, command);
     }
+    for (Consumer<Command> action : m_initActions) {
+      action.accept(command);
+    }
+
```

Comment:
The new version here is slightly different than the C++ version (CommandScheduler.cpp lines 145-153); the C++ version puts the requirements logic after the action calls.  If the order makes a difference, we should fix C++ too, if it doesn't let's at least make them consistent (probably by swapping the order here).